tests/std/tests/P0009R18_mdspan_layout_stride/test.cpp

```diff
@@ -164,12 +164,361 @@ constexpr void check_members(extents<IndexType, Extents...> ext, const array<int
+    static_assert(!is_constructible_v<Mapping, dextents<int, 2>, array<int, 3>>);
+    static_assert(!is_constructible_v<Mapping, extents<int, 2, 3>, span<int, 2>>);
+    static_assert(!is_constructible_v<Mapping, extents<int, 3, 3>, span<int, 3>>);
+    static_assert(!is_constructible_v<Mapping, dextents<int, 2>, span<int, 2>>);
+    static_assert(!is_constructible_v<Mapping, dextents<int, 2>, span<int, 3>>);
+    static_assert(!is_constructible_v<Mapping, extents<int, 3, 3>, array<NonConvertibleToAnything, 2>>);
+    static_assert(!is_constructible_v<Mapping, extents<int, 3, 3>, span<NonConvertibleToAnything, 2>>);
+    static_assert(!is_constructible_v<Mapping, extents<int, 3, 3>, array<ConvertibleToInt<int, IsNothrow::no>, 2>>);
+    static_assert(!is_constructible_v<Mapping, extents<int, 3, 3>, span<ConvertibleToInt<int, IsNothrow::no>, 2>>);
+}
+
+constexpr void check_construction_from_other_mappings() {
+    { // Check construction
+        using Mapping = layout_stride::mapping<extents<int, 4, 4>>;
+        static_assert(is_constructible_v<Mapping, layout_left::mapping<extents<int, 4, 4>>>);
+        static_assert(is_constructible_v<Mapping, layout_right::mapping<extents<int, 4, 4>>>);
+        static_assert(is_constructible_v<Mapping, layout_left::mapping<extents<long, 4, 4>>>);
+        static_assert(is_constructible_v<Mapping, layout_right::mapping<extents<long, 4, 4>>>);
+        static_assert(is_constructible_v<Mapping, layout_stride::mapping<extents<long, 4, 4>>>);
+    }
+
+    { // Check invalid construction
+        using Mapping = layout_stride::mapping<extents<int, 4, 4>>;
+        static_assert(!is_constructible_v<Mapping, layout_left::mapping<extents<int, 4, 3>>>);
+        static_assert(!is_constructible_v<Mapping, layout_right::mapping<extents<int, 3, 4>>>);
+        static_assert(!is_constructible_v<Mapping, layout_left::mapping<extents<long, 4, 4, 4>>>);
+        static_assert(!is_constructible_v<Mapping, layout_right::mapping<extents<long, 3, 3>>>);
+        static_assert(!is_constructible_v<Mapping, layout_stride::mapping<extents<int, 4, 4, 3>>>);
+        // TRANSITION, Check other kinds of invalid construction (requires new helper types)
+    }
+
+    { // Check construction from layout_left::mapping
+        layout_left::mapping<dextents<int, 3>> left_mapping{dextents<int, 3>{4, 3, 2}};
+        layout_stride::mapping<dextents<int, 3>> strided_mapping{left_mapping};
+        assert(ranges::equal(strided_mapping.strides(), array{1, 4, 12}, CmpEqual{}));
+    }
+
+    { // Check construction from layout_right::mapping
+        layout_right::mapping<dextents<int, 3>> right_mapping{dextents<int, 3>{4, 3, 2}};
+        layout_stride::mapping<dextents<int, 3>> strided_mapping{right_mapping};
+        assert(ranges::equal(strided_mapping.strides(), array{6, 2, 1}, CmpEqual{}));
+    }
+}
+
+constexpr void check_required_span_size() {
+    { // Check [mdspan.layout.stride.expo]/1.1: Ext::rank() == 0

```

Comment:
Suggestion for followup: It would be a good idea to cite the WP next to every StableName/ParagraphNumber.